src/libraries/Common/tests/System/Security/Cryptography/MLKemTests.Keys.cs

```diff
@@ -0,0 +1,698 @@
+using Xunit;
+
+namespace System.Security.Cryptography.Tests
+{
+    public static partial class MLKemTests
+    {
+        [ConditionalTheory(typeof(MLKem), nameof(MLKem.IsSupported))]
+        [MemberData(nameof(MLKemAlgorithms))]
+        public static void Generate_Roundtrip(MLKemAlgorithm algorithm)
+        {
+            using MLKem kem = MLKem.GenerateKey(algorithm);
+            Assert.Equal(algorithm, kem.Algorithm);
+
+            Span<byte> seed = stackalloc byte[MLKem.PrivateSeedSizeInBytes];
+            seed.Clear();
+
+            kem.ExportPrivateSeed(seed);
+            Assert.True(seed.ContainsAnyExcept((byte)0));
+
+            using MLKem kem2 = MLKem.ImportPrivateSeed(algorithm, seed);
+            Span<byte> seed2 = stackalloc byte[MLKem.PrivateSeedSizeInBytes];
+            kem2.ExportPrivateSeed(seed2);
+            AssertExtensions.SequenceEqual(seed, seed2);
+        }
+
+        [ConditionalFact(typeof(MLKem), nameof(MLKem.IsSupported))]
+        public static void Generate_NistVectors()
+        {
+            foreach (MLKemGenerateTestVector vector in MLKemGenerateTestVectors)
+            {
+                using MLKem kem = MLKem.ImportPrivateSeed(vector.Algorithm, vector.Seed.HexToByteArray());
+                Assert.Equal(vector.Algorithm, kem.Algorithm);
+
+                byte[] decapsKey = new byte[vector.Algorithm.DecapsulationKeySizeInBytes];
+                kem.ExportDecapsulationKey(decapsKey);
+                AssertExtensions.SequenceEqual(vector.DecapsulationKey.HexToByteArray(), decapsKey);
+
+                byte[] encapsKey = new byte[vector.Algorithm.EncapsulationKeySizeInBytes];
+                kem.ExportEncapsulationKey(encapsKey);
+                AssertExtensions.SequenceEqual(vector.EncapsulationKey.HexToByteArray(), encapsKey);
+            }
+        }
+
+        [ConditionalTheory(nameof(IsNotSupported))]
+        [MemberData(nameof(MLKemAlgorithms))]
+        public static void Generate_NotSupported(MLKemAlgorithm algorithm)
+        {
+            Assert.Throws<PlatformNotSupportedException>(() => MLKem.GenerateKey(algorithm));
+        }
+
+        [ConditionalFact(typeof(MLKem), nameof(MLKem.IsSupported))]
+        public static void ImportEncapsulationKey_Roundtrip()
+        {
+            foreach (MLKemGenerateTestVector vector in MLKemGenerateTestVectors)
+            {
+                byte[] encapsulationKeyBytes = vector.EncapsulationKey.HexToByteArray();
+                using MLKem kem = MLKem.ImportEncapsulationKey(vector.Algorithm, encapsulationKeyBytes);
+                Assert.Equal(vector.Algorithm, kem.Algorithm);
+
+                byte[] exportedEncapsulationKey = new byte[vector.Algorithm.EncapsulationKeySizeInBytes];
+                kem.ExportEncapsulationKey(exportedEncapsulationKey);
+                AssertExtensions.SequenceEqual(encapsulationKeyBytes, exportedEncapsulationKey);
+            }
+        }
+
+        [ConditionalTheory(nameof(IsNotSupported))]
```

Comment:
Personally, I'm annoyed at how many "skipping test [blah] because [blah]" the test run prints.

I vaguely wonder if we should move the "not supported" tests to a different class and give it the class level attribute, and similarly make these have a class level attribute saying supported is required.

It's a minor thing, though, so can always do it later.

Reply:
Fair. I'll do it when we get to the "okay now make it work for `MLKemOpenSsl` and `MLKemCng`".